### PR TITLE
Add frost season indices

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fulfills a need for an Ouranos project
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
    - Adds indicators `frost_season_length` and `first_day_above` (both using tasmin). 
    - Also, transforms run length's `run_length_with_dates` to `season_length`. The difference being that the `date` argument is now optional. Passing a date adds the restriction that this date be in the run. I understood "season" as meaning that the run needs to have a minimum length (`window`) *and*  may include breaks shorter than `window`. The last part is the difference between this and the simple `run_length`.
    - Moved the `consecutive_frost_days` _indice_   to `maximum_consecutive_frost_days` and made the threshold a parameter, in order to harmonize the naming with `maximum_consecutive_dry_days` and others. The `consecutive_frost_days` keeps its name (so the change is not as breaking) but now has a `thresh` parameter.

* **Does this PR introduce a breaking change?** 
Yes.
    - `xc.indices.run_length.run_length_with_dates` -> `xc.indices.run_length.season_length`. Also argument `date` is now optional and the defautl changes from "07-01" to `None`.
    - `xc.indices.consecutive_frost_days` -> `xc.indices.maximum_consecutive_frost_days` 

* **Other information**:
Each time I add such indices it's the same realization : we have many similar indices with completely different names. And many indices that could easily be generalized, but have very specific names...

### CODE REVIEW

File: tests/test_indices.py [hunk 1]
```diff
@@ -152,18 +152,18 @@ def test_simple(self, tas_series):
 class TestConsecutiveFrostDays:
```
Comment (Zeitsperre) [inline tests/test_indices.py:152]:
```suggestion
class TestMaxConsecutiveFrostDays:
```

**Review by Zeitsperre — APPROVED** [review]
Looks much cleaner. Nice addition!

**Comment by aulemahal** [comment]
I refactored the `History.rst` file by splitting "new features" and "new indicators", I thought the indicators ought to be more obvious. @Zeitsperre

File: HISTORY.rst [hunk 2]
```diff
@@ -13,6 +13,18 @@ Breaking changes
...
+* `xc.indices.consecutive_frost_days` becomes `xc.indices.maximum_consecutive_frost_days`.
+
+New indicators
+~~~~~~~~~~~~~~
+* `atmos.max_pr_intensity` acting on hourly data.
+* `atmos.wind_vector_from_speed`, also the `wind_speed_from_vector` now also returns the "wind from direction".
+* Richards-Baker flow flashiness indicator (`xclim.land.rb_flashiness_index`)
+* `atmos.max_daily_temperature_range`
+* `atmos.cold_spell_frequency`
+* `atmos.tg_min` and `atmos.tg_max`
```
Comment (Zeitsperre) [inline HISTORY.rst:26]:
```suggestion
* Richards-Baker flow flashiness indicator (`xclim.land.rb_flashiness_index`).
* `atmos.max_daily_temperature_range`.
* `atmos.cold_spell_frequency`.
* `atmos.tg_min` and `atmos.tg_max`.
```

File: xclim/indices/_threshold.py [hunk 3]
```diff
@@ -496,13 +499,80 @@ def growing_season_length(
...
+    Parameters
+    ----------
+    tasmin : xarray.DataArray
+      Minimum daily temperature [℃] or [K].
+    thresh : str
+      Threshold temperature on which to base evaluation [℃] or [K]. Default: '0.0 degC'.
+    window : int
+      Minimum number of days with temperature below threshold to mark the beginning and end of frost season.
+    freq : str
+      Resampling frequency. Default: "YS".
```
Comment (Zeitsperre) [inline xclim/indices/_threshold.py:533]:
```suggestion
      Resampling frequency. Default: "AS-JUL".
```

File: xclim/indices/_threshold.py [hunk 4]
```diff
@@ -496,13 +499,80 @@ def growing_season_length(
...
+    WARNING: The default freq value is valid for the northern hemisphere.
+
+    Parameters
+    ----------
+    tasmin : xarray.DataArray
+      Minimum daily temperature [℃] or [K].
+    thresh : str
+      Threshold temperature on which to base evaluation [℃] or [K]. Default: '0.0 degC'.
+    window : int
+      Minimum number of days with temperature below threshold to mark the beginning and end of frost season.
```
Comment (Zeitsperre) [inline xclim/indices/_threshold.py:531]:
```suggestion
      Minimum number of days with temperature below threshold to mark the beginning and end of frost season. Default: 5.
```

File: xclim/indices/_threshold.py [hunk 5]
```diff
@@ -591,6 +663,52 @@ def first_day_below(
...
+    over a given number of days, limited to a starting calendar date.
+
+    WARNING: The default date and freq are valid for the northern hemisphere.
+
+    Parameters
+    ----------
+    tasmin : xarray.DataArray
+      Minimum daily temperature [℃] or [K].
+    thresh : str
+      Threshold temperature on which to base evaluation [℃] or [K]. Default '0 degC'.
```
Comment (Zeitsperre) [inline xclim/indices/_threshold.py:686]:
```suggestion
      Threshold temperature on which to base evaluation [℃] or [K]. Default: '0 degC'.
```

**Review by Zeitsperre — APPROVED** [review]

**Comment by aulemahal** [comment]
After discussion with @tlogan2000, looks like `frost_season` _should_ include the `mid_date` option after all. I kept the option not to give it, but it now defaults to "01-01", which makes sense in the Northern Hemisphere with  `freq='AS-JUL'`.

**Pull request merged**